### ISSUE TITLE
Unable to reconnect Switchbot Curtain to Homey Pro

### ISSUE THREAD

**Issue by Arjan-vander-Meer** [AdyRock/com.switchbot#45]
After I bought a new switchbot curtain I was able to pair it to homey within seconds. Because it seemed not to work I decided to delete the device and add it again but Homey won't find the device, whatever I try. It says no new devices found. I tried the following things:

-restart homey
-remove the Swtichbot app, restart homey and reinstall the app
- factory reset the Switchbot multiple times by pressing the button for 45 seconds
- Installed the switchbot app to my phone and it pairs instantly
- disabled bluetooth on my phone and reset the Switchbot to factory settings again
- I've tried to connect the device in the Homey developer tools
- I moved closer to the Homey Pro and tried to connect
- Search fora and Google for a solution to this problem

Whatever I do I get the message "No new devices found" or in dutch "Er zijn geen nieuwe apparaten gevonden".

**Comment by Arjan-vander-Meer** [comment]
Hi Adrian,

 

I’m connecting via BLE and the first time I connected the device, it connected instantly. Really within a few seconds the device was up and running. Because I read I had to use the phone app to calibrate the device. After I calibrated the device I couldn’t connect anymore. 

In the system settings it says the Homey Model ID is Homey4d. This doesn’t sound good?

 

Kind regards,

Arjan van der Meer

 

Van: Adrian Rockall ***@***.***> 
Verzonden: dinsdag 22 november 2022 16:58
Aan: AdyRock/com.switchbot ***@***.***>
CC: Arjan-vander-Meer ***@***.***>; Author ***@***.***>
Onderwerp: Re: [AdyRock/com.switchbot] Unable to reconnect Switchbot Curtain to Homey Pro (Issue #45)

 

Are you connecting via BLE or a Switchbot hub?
If you look in the developer portal under System, what does it show for homeyModelId?
My homey4d has problems with BLE and I believe that is common. Homey3d seems to be much better.

**Comment by AdyRock** [comment]
Is your Homey on the latest firmware?
Mine is a bit better since the last update but still not as good as the homey3d.

**Comment by Arjan-vander-Meer** [comment]
Hi Adrian,

 

Yes, it’s the latest stable version (8.08). I just tried updating but it says there are no new versions available.

 

Kind regards,

Arjan van der Meer

 

Van: Adrian Rockall ***@***.***> 
Verzonden: dinsdag 22 november 2022 18:17
Aan: AdyRock/com.switchbot ***@***.***>
CC: Arjan-vander-Meer ***@***.***>; Author ***@***.***>
Onderwerp: Re: [AdyRock/com.switchbot] Unable to reconnect Switchbot Curtain to Homey Pro (Issue #45)

 

Is your Homey on the latest firmware?
Mine is a bit better since the last update but still not as good as the homey3d.

**Comment by philipostli** [comment]
Hey. I have these problems too on my Homey Bridge. First time it worked great. I removed ot from Homey and now I can not find any devices. Bought a new one today and it wont find that either. I suspect there is a problem with the account. I will try and set up a new account and see if the Homey recognizes any devices again.

**Comment by AdyRock** [comment]
Do you have a Switchbot hub or are you using BLE?

**Comment by philipostli** [comment]
Using BLE. Not going to buy a HUB, it is the main reason I use Homey to have only one HUB. Both Bridge and Pro having problems, and it was working for both of us without problems. It can not be a BLE antenna problem? It is discovered in Developer Tools. But it says "Peripheral Not Connectable". The attribute under Details "Connectable" is false. Other devices are true. Do you know what it means? Wonder if the bot can only be connected to one single device at the time. Either through the cloud or through Homey. What do you think?

**Comment by AdyRock** [comment]
The connection should be temporary, just long enough to send the command and get an acknowledgement. The rest of the time the bot sends status information in a broadcast. 
Have you enable the password in the Switchbot app?

**Comment by philipostli** [comment]
No passwords enabled. Not sure if it is even possible, when you are loged in to the Android app. I believe the bot is locked to the account, and then only apps logged in to that same account can connect to it. That is my theory. 

I can try to do a reset on the bot. Regret that I did not connect to Homey firstly to test if the problem is not there on new devices or if it started after I connected to the Android app. 

Update: Factory reset (long press 45s) and still not working with Homey.

**Comment by AdyRock** [comment]
FYI, it's obviously not the cause of the problem but it's possible to set a password for a bot in the Switchbot Android app. 
![Screenshot_20221229_145651_SwitchBot](https://user-images.githubusercontent.com/38184396/209971097-0e76791b-f8ef-4ea6-badc-dc1edf26e18b.jpg)
How far away is the bot from the Homey hub?
If possible try putting them close together to see if they will pair. 
When you see it in the Developer tools, what is showing in the Service data section as that is where the required information comes from.

**Comment by philipostli** [comment]
Interesting, I have no such option. Running version 6.20.1. The robot is right next to Homey. There is nothing in the Services section. I can not connect to it. And when I try now I can not discover it even. Might be that I mixed it up with another device before. I checked the MAC address in the Android app, and it is not showing in developer Tools. I am not very experienced in this. But what type of devices will show here? Any reason it should not show up in the list? I find my soundbar and TV in the list, but not my headset in pairing mode.

![Screenshot_2022-12-29-16-37-34-60_4c78499a84130131c925091a7f26a9c9](https://user-images.githubusercontent.com/55027973/209975957-ee699a45-f099-4795-8d7f-621423977bed.jpg)

**Comment by AdyRock** [comment]
If you pull the bridge power for a few seconds and then try to pair it again does that help?
I must admit I'm very disappointed with the BLE on the bridge and the 4d Homey. When I developed the app on my 3d Homey everything was OK.

**Comment by philipostli** [comment]
Nothing works. Have you tested it with bridge? I find it odd it is not listed in developer tools. Seems it can not be discovered at all. But I am sure it was up an running the first time I tried it. I bought it in october. Wonder if something happened in the last update? Perhaps a firmware issue? If someone have it up and running on another Homey please let me know. I will concider switching to another Homey model. Perhaps the newest Pro.

**Comment by AdyRock** [comment]
This is the difference between my bridge and my Homey Pro, which are sitting right next to each other. Basically the bridge is rubbish for BLE.
![image](https://user-images.githubusercontent.com/38184396/210863979-60bf8ceb-cba7-410c-ae70-9786a7f64daa.png)
![image](https://user-images.githubusercontent.com/38184396/210863790-6f7b5243-e4fd-4f29-a24d-3d6bb22cf8f6.png)

**Comment by philipostli** [comment]
Wow. That is a huge deal. I guess the Bridge will be switched out eventually. The new Pro seems to have alot better antennas. I'm curious. Do you work for SwitchBot or have you sold this app to them? I am interessted in contributing myself to some integrations that I see missing, and wonder how you got yours official. Cause the cloud app store is exclusively for manufactorers themselves right?

**Comment by AdyRock** [comment]
No, I don't work for them, at least not employed. I originally wrote the app and then transferred it to them but continue to maintain it. I have direct contact with the owners personal secretary, which means I get a bit more support and they have provided me with some free devices to use for development. 

Hopefully, Athom will be able to sort out the bridge eventually as its gotten worse with recent firmware updates. On the other hand the Pro 4d seems to have gotten better.

**Comment by philipostli** [comment]
Thanks for your replies Adrian! This is inspiring. Another reason for upgrading is of course all the other capabilities. I read that your Smart Things community app is much better that the official Samsung app. Keep up the good work!
